Userland/Libraries/LibWeb/Layout/TextNode.cpp

```diff
@@ -279,6 +279,477 @@ static String apply_math_auto_text_transform(String const& string)
     return MUST(builder.to_string());
 }
 
+// https://www.w3.org/TR/css-text-3/#full-width
+static ErrorOr<String> apply_full_width_text_transform(String const& string)
+{
+    // The lookup table is parsed from `UnicodeData.txt` file from:
+    // https://www.unicode.org/reports/tr44/#Character_Decomposition_Mappings
+    auto map_code_point_to_fullwidth = [](u32 code_point) -> u32 {
```

Comment:
Haven't reviewed the whole PR yet, but this stuck out to me - we should actively try to avoid defining such large Unicode property tables 😅 

We have ICU available in LibUnicode. ICU provides `icu::Transliterator` that should work for this purpose. I think in terms of API, we can add `String::to_fullwidth()` (declared in AK/String.h and defined in LibUnicode/String.cpp (yes that's weird, but that's what we currently have to do to use ICU data)).

Something like:
```c++
#include <unicode/translit.h>

String String::to_fullwidth() const
{
    UErrorCode status = U_ZERO_ERROR;

    auto transliterator = adopt_own_if_nonnull(icu::Transliterator::createInstance("Halfwidth-Fullwidth", UTRANS_FORWARD, status));
    VERIFY(Unicode::icu_success(status));

    auto icu_string = Unicode::icu_string(bytes_as_string_view());
    transliterator->transliterate(icu_string);

    return Unicode::icu_string_to_string(icu_string);
}
```

That's a bit allocation heavy, but `icu::Transliterator` doesn't seem to have nicer APIs that let us avoid converting back and forth between `icu::UnicodeString`.

Reply:
Makes sense, thanks for the feedback.